### DISCUSSION TITLE
ImageMagick performance degrades in recent versions when converting PDF pages to TIFF images

### DISCUSSION THREAD

**Discussion by wjiangot** [ImageMagick/ImageMagick#4284] [Help]
ImageMagick performance seems degraded considerably and continuously from version 6.7 to 7.0.10. In a test done on Linux, use the following command -

time ./convert -background white -extent 0x0 -define pdf:use-cropbox=true -quiet -dither FloydSteinberg -units PixelsPerInch -density 300.0x300.0 -compress Group4 -colorspace GRAY -monochrome -type Bilevel ./test.pdf[5] ./test.TIFF

the time used for several ImageMagick versions are like this:

6.7.3-6   1.058 seconds
7.0.7   2.133 seconds
7.0.9-16   4.72 seconds
7.0.10-56   2.867 seconds

The data above was obtained in a Linux box but similar results can be found in Windows too. Ignoring the large regression found in 7.0.9 (which is corrected in later versions), the trend is still getting slower and slower, from version 6.7.3 to 7.0.7 to 7.010. Is there plan at work (or some command line tricks) to gain the performance back to the level of 6.7.3 (or even better)?

**Comment by snibgo** [comment]
On general performance tips, see https://github.com/ImageMagick/ImageMagick/discussions/4224

What is the complete text output from "convert -version" in your v6 and v7 versions? Perhaps your v6 is integer (non-HDRI) but your v7 is floating-point (HDRI).

**Comment by fmw42** [comment]
Suggestion:  Try to figure out which command is the one that might be slowing down rather than a long string of commands. Find the simplest command that seems to affect the timing.
